### DISCUSSION TITLE
[Question & Enhancement] Why does CameraView not use the already existing CameraView on Android?

### DISCUSSION THREAD

**Discussion by Braini01** [xamarin/XamarinCommunityToolkit#457] [Q&A] [answered]
I was wondering why the Android implementation of the CameraView is not using the already existing [Android CameraView](https://developer.android.com/reference/androidx/camera/view/CameraView) and instead uses a rather handcrafted and complex implementation. Is there a certain feature in the Toolkit CameraView that could not be implemented by using the Android CameraView or would that approach not be flexible enough for future extensions?
If there is no specific reason, i believe a lot of the still [open Android CameraView Bugs/Issues](https://github.com/xamarin/XamarinCommunityToolkit/issues?q=is%3Aissue+is%3Aopen+label%3A%22a%2FCameraView+%3Acamera_flash%3A%22+label%3Ap%2Fandroid) could be fixed/solved by implementing the Toolkit CameraView using the Android CameraView. Additionally it probably would simplify the code significantly.

**Comment by bmacombe** [comment]
@pictos  what do you think?

**Comment by pictos** [comment]
@bmacombe I think that we can try!

**Comment by pictos** [comment]
Ops I just saw that this CameraView is on `AndroidX` namespace, so that can be an issue because we need to support devs that aren't on AndroidX yet.

**Comment by jfversluis** [comment]
That would be amazing! But as pointed out by @pictos that is only supported by AndroidX :( and we want to support the current Android support packages as well. So, unless we are overlooking something, we are stuck to creating our own for the time being.

**Comment by Braini01** [comment]
It seems the code for AndroidX CameraView can be found [here](https://androidx.tech/artifacts/camera/camera-view/1.0.0-alpha18-source/androidx/camera/view/CameraView.java.html). I could probably rewrite the XamarinCommunityToolkit Android CameraView based on this. But is it allowed legally? The code seems to be licensed with Apache 2.0 license.

**Comment by pictos** [comment]
@Braini01 thanks for your interest, but I'm already rewriting the CameraView control for Android. Any way this code will be a good reference as well, thanks for that. You can track my progress [here](https://github.com/xamarin/XamarinCommunityToolkit/tree/pj/cameraViewAndroid).

And also, enhancements are more than welcome, if you want to contribute❣

**Comment by acuntex** [comment]
@pictos Are you sure you still need to support pre AndroidX?

I recently (August 24th) got an email by the google play console, telling me that starting from November 2nd 2020 they only accept apps that target Android 10 (Api level 29).

See here:
https://developer.android.com/distribute/best-practices/develop/target-sdk.html

> Note: From 2 November 2020, app updates must target Android 10 (API level 29) or higher.

**Comment by jfversluis** [comment]
Thanks @acuntex yeah we are seeing if that is something we still want to do or not :) thanks for the input!

**Comment by pictos** [comment, marked as answer]
For now, we will not migrate to AndroidX because it's in beta. When Google releases a stable version we can add it to our backlog.
